src/consensus_cabinet.erl

```diff
@@ -56,6 +50,15 @@ compose(Label, SeatShares) when ?IS_PVC(Label) ->
                             lists:member(CenterPty, party_names(Cab)) ],
     just_party_names(WithCtrPty).
 
+% This case requires knowledge of all parties
+-spec compose(label(), [party_result()], [party_name()]) -> [cabinet()].
+compose(Label, SeatShares, AllParties) when ?IS_MCW(Label) ->
+    WithSeats   = winning_coalitions(SeatShares),
```

Comment:
Without all the context or plans for the future... I'm wondering if a map can be returned from winning_coalitions/1 so that we could do less list iterations?

Reply:
Not in a way that I can reason about. It's the set of all coalitions that could win (i.e., command a majority), that then get filtered depending on the criteria for specific cases. What keys and values were you thinking of?

Reply:
I'd have to look. I just saw the number of "going over" things and wondered if that could be reduced. As stated, you have all the context, but if you give me some time, I can look deeper.

Reply:
OK. Thanks. Since that would be a new refactoring topic in itself, I'll merge and potentially follow up.

Reply:
Totally... it's an optimization, not part of "make it work?